lib/ansible/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,464 @@
+      sample: ['UPDATE_PROFILE']
+    attributes:
+      description: Attributes applied to this user
+      type: dict
+      returned: always
+      sample:
+        attr1: ["val1", "val2", "val3"]
+    email:
+      description: Email of the user
+      type: str
+      returned: always
+      sample: john@test.com
+    access:
+      description: A dict describing the accesses you have to this user based on the credentials used.
+      type: dict
+      returned: always
+      sample:
+        manage: true
+        manageMembership: true
+        view: true
+'''
+
+from ansible.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+    meta_args = dict(
+        state=dict(default='present', choices=['present', 'absent']),
+        realm=dict(default='master'),
+        id=dict(type='str'),
+        # override auth_username from keycloak_argument_spec to remove username alias
+        auth_username=dict(type='str', required=True),
+        username=dict(type='str'),
+        email=dict(type='str'),
+        emailVerified=dict(type='bool', aliases=['email_verified']),
+        enabled=dict(type='bool'),
+        firstName=dict(type='str', aliases=['first_name']),
+        lastName=dict(type='str', aliases=['last_name']),
+        credentials=dict(type='list', no_log=True, elements='dict'),
+        requiredActions=dict(type='list', aliases=['required_actions'], elements='str',
+                             choices=['VERIFY_EMAIL', 'UPDATE_PROFILE', 'CONFIGURE_TOTP', 'UPDATE_PASSWORD',
+                                      'terms_and_conditions', 'update_user_locale']
+                             ),
+        attributes=dict(type='dict')
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([['id', 'username']]))
+
+    result = dict(changed=False, msg='', diff={}, user='')
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(
+            base_url=module.params.get('auth_keycloak_url'),
+            validate_certs=module.params.get('validate_certs'),
+            auth_realm=module.params.get('auth_realm'),
+            client_id=module.params.get('auth_client_id'),
+            auth_username=module.params.get('auth_username'),
+            auth_password=module.params.get('auth_password'),
+            client_secret=module.params.get('auth_client_secret'),
+        )
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+    kc = KeycloakAPI(module, connection_header)
+
+    realm = module.params.get('realm')
+    state = module.params.get('state')
+    gid = module.params.get('id')
```

Comment:
Should this be 'uid' instead of 'gid'? Assuming that was gid in keycloak_group.

Reply:
Indeed! 
done.